thermal_base/thermal_base.py

```diff
@@ -0,0 +1,1183 @@
+            height (int, optional): Height of the colourbar canvas to be used. If not passed, takes the height of
+                the thermal image. Defaults to None.
+
+        Returns:
+            np.ndarray: A colourbar of the required height with temperature values labelled
+        """
+        min_temp = self.global_min_temp if min_temp is None else min_temp
+        max_temp = self.global_max_temp if max_temp is None else max_temp
+
+        # Build a 255 x 21 colour bar, and create an image to append to the main image
+        cb_gray = np.arange(255, 0, -1, dtype=np.uint8).reshape((255, 1))
+        cb_gray = np.tile(cb_gray, 21)
+
+        # If the image height happens to be smaller than 255 pixels, then reduce the colour bar accordingly
+        # if self.thermal_np.shape[0] < 255:
+        #     cb_gray = cv.resize(
+        #         cb_gray, (min(21, int(self.thermal_np.shape[1] * 0.05)), int(self.thermal_np.shape[0] * 0.75))
+        #     )
+
+        cb_color = cv.applyColorMap(cb_gray, cmap) if cmap is not None else cv.cvtColor(cb_gray, cv.COLOR_GRAY2BGR)
+
+        # Decide layout of the colourbar canvas
+        cb_canvas_height_slice = height if height is not None else self.thermal_np.shape[0]
+        cb_canvas = np.zeros((cb_canvas_height_slice, cb_color.shape[1] + 30, 3), dtype=np.uint8)
+        cb_canvas_start_height = cb_canvas.shape[0] // 2 - cb_color.shape[0] // 2
+        cb_height_slice = slice(cb_canvas_start_height, cb_canvas_start_height + cb_color.shape[0])
+
+        # Populate the canvas with the colourbar and relevant text
+        cb_canvas[cb_height_slice, 10 : 10 + cb_color.shape[1]] = cb_color
+        min_temp_text = (5, cb_canvas_start_height + cb_color.shape[0] + 30)
+        max_temp_text = (5, cb_canvas_start_height - 20)
+        cv.putText(cb_canvas, str(round(min_temp, 2)), min_temp_text, cv.FONT_HERSHEY_PLAIN, 1, (255, 0, 0), 1, 8)
+        cv.putText(cb_canvas, str(round(max_temp, 2)), max_temp_text, cv.FONT_HERSHEY_PLAIN, 1, (0, 0, 255), 1, 8)
+
+        return cb_canvas
+
+    def save_thermal_image(self, output_path):
+        """
+        Converts the floating-point temperature array into an image and saves the image
+        Takes the current `self.thermal_np` temperature array, converts it to image-friendly uint8 format. Then it
+        computes and stitches a colourbar for required scale. Finally it saves the image to the specified output path
+
+        Args:
+            output_path (str / Path): Save destination for the thermal image
+        """
+        # Generate a colour mapped image using default colour map
+        cmapped_thermal_img = ThermalImageHelpers.get_cmapped_temp_image(self.thermal_np, colormap=self.cmap)
+        cb_canvas = self.generate_colorbar(cmap=self.cmap)
+        cmapped_thermal_img = cv.hconcat((cmapped_thermal_img, cb_canvas))
+        cv.imwrite(str(output_path), cmapped_thermal_img)
+
+
+class ThermalImageHelpers:
```

Comment:
Possible modification (you can evaluate and decide): Since this class is composed entirely of static methods, you can instead just shift the functions to a separate `utils.py` file, and have the functions defined directly there without a class grouping.